posthog/clickhouse/migrations/README.md

```diff
@@ -0,0 +1,54 @@
+## About migrations
+
+Not all migrations are intended to run on all nodes every time, because of the topologies we run. Some nodes are intended to only perform compute operations and do not contain sharded tables.
+
+Some tables are meant to run only on a couple of nodes, like some Kafka tables, to prevent the whole cluster to run too many insertions in ClickHouse.
+
+And, in some other cases, we want to only create one table (like a unique Kafka consumer) in one node.
+
+Because of the above, take the following advice into consideration when manipulating schemas for ClickHouse.
+
+### When to run a migration ONLY on a worker node
+
+- When adding / updating a Kafka table
+- When adding / updating a materialized view
+- When adding / updating a sharded table
+- When adding / updating a distributed table used for the write path
+
+In the above cases, create a migration and call the `run_sql_with_exceptions` function with the `node_role` set to `NodeRole.WORKER`.
+
+<details>
+
+<summary>Example</summary>
+For example, the `sharded_events` table is a sharded table. Thus, it should only be added on worker nodes.
+
+Also, since to fill this table we need to consume events from Kafka, we need to run Kafka consumers on the worker nodes, which would include the materialized view and the writable distributed table. So the `kafka_events_json`, `events_json_mv` and `writable_events` tables should also be added on them.
+
+</details>
+
+
+### When to run a migration for all nodes
+
+- Basically when the migration does not include any of the above listed in the previous section.
+- When adding / updating a distributed table for reading
+- When adding / updating a replicated table
+- When adding / updating a view
+- When adding / updating a dictionary
+- And so on
+
+In the above cases, create a migration and call the `run_sql_with_exceptions` function with the `node_role` set to `NodeRole.ALL`.
+
+<details>
+
+<summary>Example</summary>
+
+Following the previous section example, the sharded events table along with the Kafka tables, materialized views and writable distributed table would be added to the worker nodes. However, the `distributed_events`, which is the table used for the read path, would be added to all nodes.
+
+</details>
+
+### The ON CLUSTER clause
```

Comment:
Extra reason for not using "ON CLUSTER" -> it put's the DDL to be run in zookeeper, the same queue as DELETE. It may get stuck and keep the cluster in inconsistent state.